backend/private-graph/graph/resolver.go

```diff
@@ -3671,3 +3673,74 @@ func IsOptOutTokenValid(adminID int, token string) bool {
 	// If the token matches the prior month's, it's valid
 	return token == Email.GetOptOutToken(adminID, true)
 }
+
+func (r *Resolver) CreateErrorTag(ctx context.Context, title string, description string) (*model.ErrorTag, error) {
+	errorTag, err := r.EmbeddingsClient.GetErrorTagEmbedding(ctx, title, description)
+
+	if err != nil {
+		log.WithContext(ctx).Error(err, "CreateErrorTag: Error creating tag embedding")
+		return nil, err
+	}
+
+	if err := r.DB.Create(errorTag).Error; err != nil {
+		log.WithContext(ctx).Error(err, "CreateErrorTag: Error creating tag")
+		return nil, err
+	}
+
+	return errorTag, nil
+}
+
+func (r *Resolver) GetErrorTags() ([]*model.ErrorTag, error) {
+	var errorTags []*model.ErrorTag
+
+	if err := r.DB.Model(errorTags).Scan(&errorTags).Error; err != nil {
+		return nil, e.Wrap(err, "500: error querying error tags")
+	}
+
+	return errorTags, nil
+}
+
+func (r *Resolver) MatchErrorTag(ctx context.Context, query string) ([]*modelInputs.MatchedErrorTag, error) {
+	stringEmbedding, err := r.EmbeddingsClient.GetStringEmbedding(ctx, query)
+
+	if err != nil {
+		return nil, e.Wrap(err, "500: failed to get string embedding")
+	}
+
+	var matchedErrorTags []*modelInputs.MatchedErrorTag
+	if err := r.DB.Raw(`
+		select error_tags.embedding <=> @string_embedding as score,
+					error_tags.id as id,
+					error_tags.title as title,
+					error_tags.description as description
+		from error_tags
+		order by score asc
+		limit 5;
+	`, sql.Named("string_embedding", model.Vector(stringEmbedding))).
+		Scan(&matchedErrorTags).Error; err != nil {
+		return nil, e.Wrap(err, "error querying nearest ErrorTag")
+	}
+
+	return matchedErrorTags, nil
+}
+
+func (r *Resolver) FindSimilarErrors(ctx context.Context, query string) ([]*model.MatchedErrorObject, error) {
```

Comment:
can this use `modelInputs.MatchedErrorObject`?

Reply:
I went back and forth on this one, and eventually created this new struct type so that I could pass out the `score`.

```go
type MatchedErrorObject struct {
	ErrorObject
	Score float64 `json:"score"`
}
```